internal/cryptosetup/cryptosetup.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cryptosetup
+
+import (
+	"bytes"
+	"encoding/base64"
+	"errors"
+
+	"github.com/google/tink/go/keyset"
+	"github.com/google/tink/go/subtle/random"
+	ariescrypto "github.com/hyperledger/aries-framework-go/pkg/crypto"
+	"github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/packer/legacy/authcrypt"
+	"github.com/hyperledger/aries-framework-go/pkg/kms"
+	"github.com/hyperledger/aries-framework-go/pkg/kms/legacykms"
+	ariesstorage "github.com/hyperledger/aries-framework-go/pkg/storage"
+	"github.com/trustbloc/edge-core/pkg/storage"
+)
+
+const (
+	masterKeyStoreName  = "masterkey"
+	masterKeyDBKeyName  = masterKeyStoreName
+	vcIDEDVIndexName    = "vcID"
+	keyIDStoreName      = "keyid"
+	keyIDDBKeyName      = keyIDStoreName
+	signingKeyStoreName = "signingkey"
+	signingKeyDBKeyName = signingKeyStoreName
+)
+
+var errKeySetHandleAssertionFailure = errors.New("unable to assert key handle as a key set handle pointer")
+
+type legacyKMSProvider struct {
+	kms legacykms.KeyManager
+}
+
+func (p legacyKMSProvider) LegacyKMS() legacykms.KeyManager {
+	return p.kms
+}
+
+// PrepareMasterKeyReader prepares a master key reader for secret lock usage
+func PrepareMasterKeyReader(kmsSecretsStoreProvider ariesstorage.Provider) (*bytes.Reader, error) {
+	masterKeyStore, err := kmsSecretsStoreProvider.OpenStore(masterKeyStoreName)
+	if err != nil {
+		return nil, err
+	}
+
+	masterKey, err := masterKeyStore.Get(masterKeyDBKeyName)
+	if err != nil {
```

Comment:
no need for else

Reply:
if err != nil {
		if errors.Is(err, ariesstorage.ErrDataNotFound) {
			masterKeyRaw := random.GetRandomBytes(uint32(32))
			masterKey = []byte(base64.URLEncoding.EncodeToString(masterKeyRaw))

			putErr := masterKeyStore.Put(masterKeyDBKeyName, masterKey)
			if putErr != nil {
				return nil, putErr
			}
		}
			return nil, err
	}